sympy/sets/sets.py

```diff
@@ -1465,12 +1465,32 @@ def reduce(args):
                 unk = [x for x in s
                        if any(other.contains(x) not in (True, False) for other in other_args)]
                 if unk:
+                    for val in other_args:
+                        if val.is_FiniteSet:
+                            # all values from val which are contained in `s`
+                            # and are non-numbers
+                            t = [x for x in s if val.contains(x) is true and not x.is_number]
```

Comment:
BTW i doubting the use of `.atoms(Symbol)` as its solution. I guess there can be a better way to do that comparison.